pennylane/_device.py

```diff
@@ -175,6 +174,18 @@ def observables(self):
         """
         raise NotImplementedError
 
+    def set_shots(self, shots):
```

Comment:
Should we also make `shots` a property, and internally use `self._shots`? This will avoid users attempting to change `shots` and causing internally consistency issues.

We could even go further, and use the `@shots.setter` decorator here, which would allow the user to use the syntax
```
dev.shots = 32
```
but still call the logic/validation in this method.